cranelift-filetests/src/function_runner.rs

```diff
@@ -67,19 +67,23 @@ impl FunctionRunner {
         let code_info = context
             .compile(self.isa.as_ref())
             .map_err(|e| e.to_string())?;
-        let code_page = MemoryMap::new(code_info.total_size as usize, &[MapOption::MapWritable])
-            .map_err(|e| e.to_string())?;
+        let mut code_page =
+            MmapMut::map_anon(code_info.total_size as usize).map_err(|e| e.to_string())?;
         let callable_fn: fn() -> bool = unsafe {
             context.emit_to_memory(
                 self.isa.as_ref(),
-                code_page.data(),
+                code_page.as_mut_ptr(),
                 relocs,
                 traps,
                 stackmaps,
             );
-            region::protect(code_page.data(), code_page.len(), Protection::ReadExecute)
-                .map_err(|e| e.to_string())?;
-            mem::transmute(code_page.data())
+            region::protect(
```

Comment:
Please use `let code_page = code_page.make_exec().map_err(|e| e.to_string())?;` instead. Maybe switch [`cranelift-simplejit`](https://github.com/CraneStation/cranelift/blob/master/cranelift-simplejit/src/memory.rs) to that method too.

Reply:
@tschneidereit said on IRC he'd prefer this to be done as a follow-up; will open a new issue for this.